### DISCUSSION TITLE
Long term plans for GSON (since it is now in maintenance mode)

### DISCUSSION THREAD

**Discussion by thahnen** [eclipse-lsp4j/lsp4j#824] [General]
Hey everyone,

I was wondering if you already scratched your head about [GSON now being in maintenance](https://github.com/google/gson?tab=readme-ov-file#gson) mode or already have some long-term plan for the future to either replace GSON or to use a fork or whatever.

From the issues I found mentioning GSON, it is currently a "use it as it works" approach, but that might only work for a short amount of time.
We rely heavily on LSP4J (and therefore GSON as well) and, knowing that many Eclipse plug-ins do the same, are just eager to know what approach you have in mind.

Best,
Tobias

**Comment by jonahgraham** [comment]
Thanks for bringing that up. 

GSON has been in maintenance mode for many years already and I had intepretted it as "only bug fixes". There have been regular bug fix releases, so I don't know what appetite there is for rewriting this part of the stack. One challenge is that the GSON API leaks out of LSP4J, so it isn't possible to simply replace GSON with (for example) [moshi](https://github.com/square/moshi) without significantly disrupting downstream consumers of LSP4J.

So, in conclusion I know of no plan to replace GSON, but we should remain vigilant to it moving from maintenance mode to EOL.
